Content.Client/SS220/Signature/SignatureControl.cs

```diff
@@ -0,0 +1,302 @@
+                var py = y + yy;
+
+                if (px < 0 || px >= Data.Width || py < 0 || py >= Data.Height)
+                    continue;
+
+                if (_isErasing)
+                {
+                    if (!Data.GetPixel(px, py))
+                        continue;
+
+                    Data.SetPixel(px, py, true);
+                }
+                else
+                {
+                    if (Data.GetPixel(px, py))
+                        continue;
+
+                    Data.SetPixel(px, py);
+                }
+
+                _dirty = true;
+            }
+        }
+    }
+
+    private void DrawLine(int x0, int y0, int x1, int y1)
+    {
+        var dx = Math.Abs(x1 - x0);
+        var sx = x0 < x1 ? 1 : -1;
+        var dy = -Math.Abs(y1 - y0);
+        var sy = y0 < y1 ? 1 : -1;
+        var err = dx + dy;
+
+        while (true)
+        {
+            DrawPixel(x0, y0);
+
+            if (x0 == x1 && y0 == y1)
+                break;
+
+            var e2 = err * 2;
+
+            if (e2 >= dy)
+            {
+                err += dy;
+                x0 += sx;
+            }
+
+            if (e2 <= dx)
+            {
+                err += dx;
+                y0 += sy;
+            }
+        }
+    }
+
+    public void ClearSignature()
+    {
+        if (Data == null)
+            return;
+
+        Data.Clear();
+        _dirty = true;
+        SignatureChanged?.Invoke(Data);
+        InvalidateMeasure();
+    }
+
+    private void UpdateCanvas(DrawingHandleScreen handle)
+    {
+        if (CanvasSize.X <= 0 || CanvasSize.Y <= 0)
+            return;
+
+        if (Data == null)
+            return;
+
+        if (_canvas == null)
+        {
+            _canvas = _clyde.CreateRenderTarget(
+                new Vector2i((int)CanvasSize.X, (int)CanvasSize.Y),
+                new RenderTargetFormatParameters(RenderTargetColorFormat.Rgba8Srgb),
+                name: "signature");
+            _dirty = true;
+        }
+
+        if (!_dirty)
+            return;
+
+        handle.RenderInRenderTarget(_canvas,
+            () =>
+            {
+                for (var y = 0; y < Data.Height; y++)
+                {
+                    for (var x = 0; x < Data.Width; x++)
+                    {
+                        if (!Data.GetPixel(x, y))
+                            continue;
+
+                        var r = new UIBox2i(x, y, x + 1, y + 1);
+                        handle.DrawRect(r, Color.Black);
+                    }
+                }
+            },
+            Color.Transparent);
+
+        _dirty = false;
+    }
+
+    protected override void Draw(DrawingHandleScreen handle)
+    {
+        base.Draw(handle);
+
+        var size = PixelSize;
+        var rect = new UIBox2(0, 0, size.X, size.Y);
+
+        if (Editable)
+        {
+            handle.DrawRect(rect, Color.FromHex("#ffffff88"));
+            handle.DrawRect(rect, Color.Black.WithAlpha(0.4f), false);
```

Comment:
```suggestion
            handle.DrawRect(rect, Color.FromHex(SomeConstMaybeEvenPublicPropertyOne));
            handle.DrawRect(rect, Color.Black.WithAlpha(SomeConstMaybeEvenPublicPropertyTwo), false);
```

Я мейнтейнер простой:
 Не понял что за цифры - прошу убрать под константы

Reply:
MaintainerColorSmth, MainetainerColorOtherSmth